src/index.ts

```diff
@@ -156,99 +242,93 @@ export const findNextItem = (stimuli: Stimulus[], theta = 0, method = 'MFI', dee
     //findClosest requires arr is sorted by difficulty
-    const index = findClosest(arr, theta + 0.481);
+    const index = Cat.findClosest(arr, this._theta + 0.481);
     const nextItem = arr[index];
     arr.splice(index, 1);
     return {
       nextStimulus: nextItem,
       remainingStimuli: arr,
     };
-  } else if (method === 'random') {
-    const index = Math.floor(Math.random() * arr.length);
+  }
+
+  private selectorRandom(arr: Stimulus[]){
+
+    const index = Math.floor(this._rng() * arr.length);
     const nextItem = arr.splice(index, 1)[0];
     return {
       nextStimulus: nextItem,
       remainingStimuli: arr,
     };
   }
 
-  /**
-   * return a random integer between min and max
-   * @param min - The minimum of the random number range (include)
-   * @param max - The maximum of the random number range (include)
-   * @returns {number} - random integer within the range
-   */
-  function randomInteger(min: number, max: number) {
-    return Math.floor(Math.random() * (max - min + 1)) + min;
-  }
+    /**
+     * return a random integer between min and max
+     * @param min - The minimum of the random number range (include)
+     * @param max - The maximum of the random number range (include)
+     * @returns {number} - random integer within the range
+     */
+    private static randomInteger(min: number, max: number) {
+      return Math.floor(Math.random() * (max - min + 1)) + min;
+    }
 
-  /**
-   * find the item in a given array that has the difficulty closest to the target value
-   * @param arr Array<Stimulus> - an array of stimulus
-   * @param target number - ability estimate
-   * @returns {number} the index of arr
-   */
-  function findClosest(arr: Array<Stimulus>, target: number) {
-    const n = arr.length;
-    // Corner cases
-    if (target <= arr[0].difficulty) return 0;
-    if (target >= arr[n - 1].difficulty) return n - 1;
-    // Doing binary search
-    let i = 0,
-      j = n,
-      mid = 0;
-    while (i < j) {
-      mid = Math.ceil((i + j) / 2);
-      if (arr[mid].difficulty == target) return mid;
-      // If target is less than array
-      // element,then search in left
-      if (target < arr[mid].difficulty) {
-        // If target is greater than previous
-        // to mid, return closest of two
-        if (mid > 0 && target > arr[mid - 1].difficulty) return getClosest(arr, mid - 1, mid, target);
-        // Repeat for left half
-        j = mid;
-      }
-      // If target is greater than mid
-      else {
-        if (mid < n - 1 && target < arr[mid + 1].difficulty) return getClosest(arr, mid, mid + 1, target);
-        i = mid + 1; // update i
+    /**
+     * find the item in a given array that has the difficulty closest to the target value
+     * @param arr Array<Stimulus> - an array of stimulus
+     * @param target number - ability estimate
+     * @returns {number} the index of arr
+     */
+    private static findClosest(arr: Array<Stimulus>, target: number) {
```

Comment:
Maybe this should be moved out of the `Cat` class and used as a standalone function in `src/utils.ts`.